modules/ghlint-snakeyaml/src/main/kotlin/net/twisterrob/ghlint/model/SnakeWorkflow.kt

```diff
@@ -36,8 +35,8 @@ public class SnakeWorkflow internal constructor(
 			}
 			.associateBy { it.id }
 
-	override val permissions: Map<String, String>?
-		get() = node.getOptional("permissions")?.run { map.toTextMap() }
+	override val permissions: Permissions?
+		get() = node.getOptional("permissions")?.let { factory.createPermissions(it) }
```

Comment:
These duplicate changes in Job/Workflow are interesting, this is the third time (env, permissions, defaults, [name, if we force it]). Do you think it would be worth a shared super interface? or if it's a single interface it would go against interface segregation?

(out of scope for this PR, just chattin')